### PR TITLE
Improve consumer state handling.

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>
 
/cc @nats-io/core

### CODE REVIEW

File: server/jetstream_cluster_test.go [hunk 1]
```diff
@@ -11206,6 +11206,71 @@ func TestJetStreamRemovedPeersAndStreamsListAndDelete(t *testing.T) {
...
+	c := sc.clusterForName(pcn)
+	for _, s := range c.servers {
+		sd := s.JetStreamConfig().StoreDir
+		s.Shutdown()
+		removeDir(t, sd)
+		s = c.restartServer(s)
+		c.waitOnServerHealthz(s)
+	}
+
+	c.waitOnConsumerLeader("$G", "T", "d")
```
Comment (matthiashanel) [inline server/jetstream_cluster_test.go:11262]: the way I read your email, don't these two lines mask the issue?
The issue happened before the consumer leader was ready, but didn't happen after.
So shouldn't one instance of the test avoid these and a much shorter test test that once `c.waitOnServerHealthz(z)` returned the consumer has a leader (vs wait for one)

**Review by matthiashanel — COMMENTED** [review]
Change itself makes sense. 
I just have one question about the test.

File: server/monitor.go [hunk 2]
```diff
@@ -2772,21 +2783,23 @@ func (s *Server) HandleHealthz(w http.ResponseWriter, r *http.Request) {
 						if !cc.isStreamCurrent(accName, sname) {
 							health.Status = "unavailable"
 							health.Error = fmt.Sprintf("JetStream stream %q for account %q is not current", sname, accName)
-							w.WriteHeader(http.StatusServiceUnavailable)
 							js.mu.RUnlock()
 							break Err
 						}
+						// Now do consumers.
+						for _, o := range stream.getConsumers() {
```
Comment (kozlovic) [inline server/monitor.go:2790]: I wonder if there was not a reason you did not add consumers to the probe when you did the original PR... but I don't recall.

**Review by kozlovic — APPROVED** [review]
LGTM

File: server/monitor.go (hunk 2 above)
Comment (derekcollison) [reply server/monitor.go:2790]: In any case was an oversight, good catch by @matthiashanel

File: server/jetstream_cluster_test.go (hunk 1 above)
Comment (derekcollison) [reply server/jetstream_cluster_test.go:11262]: Will merge but we can add on additional torture tests for state being wiped or corrupt for sure.

**Pull request merged**